# true for object present, false for object not present
bool detection